loopback/index.d.ts

```diff
@@ -0,0 +1,2585 @@
+      * @param {string} principalType The principal type.
+      * @param {string} principalId The principal ID.
+      * @param {string} model The model name.
+      * @param {string} property The property/method/relation name.
+      * @param {string} accessType The access type.
+      * @callback {() => void} callback Callback function.
+      * @param {string|Error} err The error object
+      * @param {AccessRequest} result The access permission
+      */
+
+      checkPermission(principalType: string, principalId: string, model: string, property: string, accessType: string, callback: (err: string|Error, result: AccessRequest) => void): void;
+
+      /**
+      * Check if the request has the permission to access.
+      * @options {any} context See below.
+      * @property {any[]} principals An Array of principals.
+      * @property {string|Model} model The model name or model class.
+      * @property {*} id The model instance ID.
+      * @property {string} property The property/method/relation name.
+      * @property {string} accessType The access type:
+      *   READ, REPLICATE, WRITE, or EXECUTE.
+      * @param {() => void} callback Callback function
+      */
+
+      checkAccessForContext(callback: () => void, context: {principals: any[], model: string|Model, id: any, property: string, accessType: string}): void;
+
+      /**
+      * Check if the given access token can invoke the method
+      * @param {AccessToken} token The access token
+      * @param {string} model The model name
+      * @param {*} modelId The model id
+      * @param {string} method The method name
+      * @callback {() => void} callback Callback function
+      * @param {string|Error} err The error object
+      * @param {boolean} allowed is the request allowed
+      */
+
+      checkAccessForToken(token: AccessToken, model: string, modelId: any, method: string, callback: (err: string|Error, allowed: boolean) => void): void;
+
+      /**
+      * Resolve a principal by type/id
+      * @param {string} type Principal type - ROLE/APP/USER
+      * @param {string|number} id Principal id or name
+      * @param {() => void} cb Callback function
+      */
+
+      resolvePrincipal(type: string, id: string|number, cb: () => void): void;
+
+      /**
+      * Check if the given principal is mapped to the role
+      * @param {string} principalType Principal type
+      * @param {string|*} principalId Principal id/name
+      * @param {string|*} role Role id/name
+      * @param {() => void} cb Callback function
+      */
+
+      isMappedToRole(principalType: string, principalId: string|any, role: string|any, cb: () => void): void;
+
+}
+
+/**
+* Manage client applications and organize their users.
+*
+* @property {string} id  Generated ID.
+* @property {string} name Name; required.
+* @property {string} description Text description
+* @property {string} icon string Icon image URL.
+* @property {string} owner User ID of the 
```

Comment:
If the class inherits from PersistedModel, you could express that in TypeScript: `declare class User extends PersistedModel { ... }` (or `implements` is `PersistedModel` is an interface)